app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,113 @@
 "use client";
 import PageLayout from "../../pageLayout/pageLayout";
-import React from "react";
-import SearchResults from "../../search/results";
+import React, { useState, useRef, useEffect } from "react";
 import {
   Box,
   Heading,
   HorizontalRule,
   SearchBar,
+  Menu,
+  Text,
+  Pagination,
 } from "@nypl/design-system-react-components";
+import { useSearchParams, usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import { displayResults, totalNumPages } from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
 
 export const CollectionsPage = ({ data }) => {
+  const { push } = useRouter();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const collections = data.collection; // typeof data.collection === "object" ? [data.collection] : data.collection // this breaks empty search
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  // pagination
+  const numFound = data.numFound ? data.numFound : data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  // defaults
+  let currentPage = Number(searchParams.get("page")) || 1;
+  let currentSort = searchParams.get("sort") || "date-desc";
+  let currentCollectionKeyword = searchParams.get("collection_keyword") || "";
+
+  const handleSearchSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    const queryString = createQueryStringFromObject({
+      collection_keyword: currentCollectionKeyword,
+      sort: currentSort,
+      page: currentPage,
+    });
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 2000);
+  };
+
+  // I'm not actually sure if this is necessary but it's in the research catalog
+  const handleSearchChange = (e: SyntheticEvent) => {
+    const target = e.target as HTMLInputElement;
+    currentCollectionKeyword = target.value;
+  };
+
+  // pagination
+  // Question: Do we want to introduce debouncing?
```

Comment:
Yes.

Reply:
Actually, what would we need it for?

Reply:
We don't _need_ it. It would introduce a new feature of retrieving new data when a user stops typing. https://nextjs.org/learn/dashboard-app/adding-search-and-pagination#best-practice-debouncing

Just flagging it now but perhaps this is a product decision. We can build it out later.

Reply:
A few points:

- We should definitely use debouncing where necessary. My question was mostly why do we need/want debouncing with pagination? It's typically used for text inputs so I was curious about that. Perhaps a typo but if not, how does debouncing help with pagination?
- The page does not perform a search as the user types, they must hit "search". For the current set of features, the app doesn't need debouncing. If it's needed in the future, yes, you can build it, and yes, it's a product decision.
- I'd like to get away from having to import various packages for features but `use-debounce` looks small and very well maintained.